api/tests/Unit/PoolAuthorizationScopeTest.php

```diff
@@ -285,5 +286,127 @@ public function testViewAsPlatformAdmin(): void
         ], $poolIds->toArray());
     }
 
-    // TODO: add tests for new roles as part of #10609
+    // process operator can see all except the draft pool they are not teamed with thru authorizedToView
+    public function testScopeAuthorizedToViewAsProcessOperator(): void
+    {
+        Auth::shouldReceive('user')
+            ->andReturn(User::factory()
+                ->asProcessOperator($this->poolDraft1->id)
+                ->create());
+
+        $poolIds = Pool::authorizedToView()->get()->pluck('id')->toArray();
+
+        assertEqualsCanonicalizing([
+            $this->poolDraft1->id,
+            $this->poolPublished1->id,
+            $this->poolClosed1->id,
+            $this->poolArchived1->id,
+            $this->poolPublished2->id, // poolDraft2 not present here
+            $this->poolClosed2->id,
+            $this->poolArchived2->id,
+        ], $poolIds);
+    }
+
+    // community recruiter can see all except the draft pool they are not teamed with by pool->community thru authorizedToView
+    public function testScopeAuthorizedToViewAsCommunityRecruiter(): void
+    {
+        $community = Community::factory()->create();
+        $this->poolDraft1->community_id = $community->id;
+        $this->poolDraft1->save();
+
+        Auth::shouldReceive('user')
+            ->andReturn(User::factory()
+                ->asCommunityRecruiter($community->id)
+                ->create());
+
+        $poolIds = Pool::authorizedToView()->get()->pluck('id')->toArray();
+
+        assertEqualsCanonicalizing([
```

Comment:
I feel like there's a case missing from these tests if Community Recruiter and Process Operator can see the same pools. There should be a second pool in Community 1 which is visible to Community roles but not the Process Operator.

Reply:
That seems reasonable, I added an additional community pool in the trio of tests in https://github.com/GCTC-NTGC/gc-digital-talent/pull/11273/commits/c5291790fca47d4e6f6d9cc6d3c70eddc9f43f46